File: durak-server-orleans/Durak.Server.API/Models/GameState.cs
using Durak.Server.API.Game.Entities;
using Durak.Server.API.Game.Enums;

namespace Durak.Server.API.Models;

[GenerateSerializer]
public class GameState(
    List<PlayerState> playerStates,
    int deckCardCount,
    Suit trump,
    Card trumpCard,
    Turn currentTurn,
    List<Card> cards)
{
    [Id(0)]
    public List<PlayerState> PlayerStates { get; set; } = playerStates;
    [Id(1)]
    public int DeckCardCount { get; set; } = deckCardCount;
    [Id(2)]
    public Suit Trump { get; set; } = trump;
    [Id(3)]
    public Card TrumpCard { get; set; } = trumpCard;
    [Id(4)]
    public Turn CurrentTurn { get; set; } = currentTurn;
    [Id(5)]
    public List<Card> Cards { get; set; } = cards;
}

File: durak-client-angular/src/app/models/game-state.ts
import { Card } from "./card";
import { Suit } from "./enums/suit";
import { PlayerState } from "./player-state";
import { Turn } from "./turn";

export class GameState {
    constructor(
        public playerStates: PlayerState[], 
        public deckCardCount: number,
        public trump: Suit,
        public trumpCard: Card,
        public currentTurn: Turn,
        public cards: Card[]) { }
}